### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by saracarl**

### CODE REVIEW

File: spec/fixtures/manifests/bad_not_a_hash.json [hunk 1]
```diff
@@ -0,0 +1,142 @@
+{
+	"@context": "http://iiif.io/api/presentation/2/context.json",
+	"@id": "https://data.ucd.ie/api/img/manifests/ivrla:3832",
+	"rendering": {
+		"@id": "https://digital.ucd.ie/get/ivrla:3832/content",
+		"format": "application/pdf",
+		"label": "Download as PDF"
+	},
+	"seeAlso": [
```
Comment (jpstroop) [inline spec/fixtures/manifests/bad_not_a_hash.json]: This is a bug in o'sullivan--it a list should be allowed for `seeAlso`

File: lib/iiif/service.rb [hunk 2]
```diff
@@ -196,7 +196,12 @@ def to_ordered_hash(opts={})
       export_hash
     end
 
+    class ExpectedHashError < StandardError; end
+
     def self.from_ordered_hash(hsh, default_klass=IIIF::OrderedHash)
+      unless hsh.is_a? Hash 
+        raise IIIF::Service::ExpectedHashError, "expected a hash"
```
Comment (jcoyne) [inline lib/iiif/service.rb]: Why do we need a custom error here, wouldn't `ArgumentError` suffice?

File: lib/iiif/service.rb [hunk 3]
```diff
@@ -206,26 +211,31 @@ def self.from_ordered_hash(hsh, default_klass=IIIF::OrderedHash)
...
+                new_object[new_key] << member
+                # Again, no nested arrays, right?
+              end
             end
+          else
+            new_object[new_key] = hsh[key]
           end
-        else
-          new_object[new_key] = hsh[key]
+        rescue ExpectedHashError
```
Comment (jcoyne) [inline lib/iiif/service.rb]: I'd prefer we don't do flow control with exceptions. Can we check that `hsh[key]` and `member` are hashes here?

**Comment by benwbrum** [comment]
This pull request has been replaced by #68 and may be closed.

**Comment by saracarl** [comment]
Closing this in favor of #68

**Pull request closed without merging**